SalahAdDin/strapi-plugin-rich-text
# feat: #8 add table to the editor

### What does it do?

It adds the Tiptap's table extension, toolbar button, and floating toolbar to the editor.

### Why is it needed?
Some content editors would want to add simple tables to their posts.

### How to test it?
Select an empty line, click the Table button in the toolbar, see the floating toolbar to work with rows or columns, or delete the table.

https://github.com/user-attachments/assets/1bf70ec9-6eaf-48b3-bd47-4ab6a5237f88

### Notes

- It requires merging the following the [PR above](https://github.com/konstantinmuenster/strapi-plugin-rich-text/pull/19).
- We just chose the more required tools(listed [here](https://medium.com/@zreejur/tiptap-rich-editor-its-really-rich-f8da7659a5dc)), but you can find all possible controls for the table in the [documentation](https://tiptap.dev/docs/editor/extensions/nodes/table), and choose how to add to the TableToolbar.
-  We used simple icons, they could be ugly; feel free to introduce better icons for them, and update the respective components; don't add icon libraries.
